File: Unit1/c-programs/Programa181/v2/productor.c
#include <stdio.h>
#include "fichero.h"

int main(int argc,char **argv){
  int contador;
  contador=leovalor();
  printf(" Productor: Lectura=%d \n",contador);
  sleep(5);
  contador++;
  escribovalor(contador);
  printf(" Productor: Escritura=%d \n",contador);
}

File: Unit1/java-programs/Programa 1.8.3/src/Productor.java

public class Productor extends Thread{
  public Productor(String Nombre) {
	if (Nombre!=null) setName(Nombre);
	start();
  }
  public void run(){
	int contador;
	contador=Fichero.leovalor();
	System.out.printf(" Productor: Lectura=%d \n",contador);
	try {sleep(500);
	} catch (InterruptedException e) {};
	contador++;
	Fichero.escribovalor(contador);
	System.out.printf(" Productor: Escritura=%d \n",contador);
  }
}
